modin/pandas/groupby.py

```diff
@@ -690,10 +712,14 @@ def _default_to_pandas(self, f, *args, **kwargs):
             by = self._by
 
         def groupby_on_multiple_columns(df, *args, **kwargs):
+            by = kwargs.pop("__by")
             return f(
-                df.groupby(by=by, axis=self._axis, **self._kwargs), *args, **kwargs
+                df.groupby(by=by, axis=self._axis, **self._kwargs), *args, **kwargs,
             )
 
+        # Add by to kwargs, so it can get automatically converted to Pandas
+        # when necessary:
+        kwargs["__by"] = by
```

Comment:
why `__by`? maybe to make it more explicit we should do 
```python
return self._df._default_to_pandas(groupby_on_multiple_columns, by=by, *args, **kwargs)
```
or this was done to avoid some name conflicts?

Reply:
The goal was to avoid name conflicts, yes. There may not be any, but didn't want to risk it.

Another alternative is to continue to have `by` as it was before, passed via closure, and to explicitly convert it with `try_convert_to_pandas()` or whatever it's called.

Reply:
yeah, I see, let's then keep passing `by` via closure with converting it via `try_cast_to_pandas`

Reply:
Done.